zero/src/lib.rs

```diff
@@ -1,3 +1,5 @@
+#![feature(array_try_from_fn)]
+
```

Comment:
was it just for convenience to remove the macro in `prover_state/mod.rs`?
In general, I'd tend to discourage unstable features (I think that's also @0xaatif's opinion?) unless these are in a fairly mature state already / on a path to be made stable soon / strongly needed (i.e. performance reasons for instance).

Reply:
Thanks for the reviews! I have removed the dependency on this feature.